modules/newrelic-http-exporter/src/main/scala/io/janstenpickle/trace4cats/newrelic/Endpoint.scala

```diff
@@ -1,16 +1,18 @@
 package io.janstenpickle.trace4cats.newrelic
 
+import org.http4s.Uri
+
 sealed trait Endpoint {
-  def url: String
+  def uri: Uri
```

Comment:
This is a tiny but incompatible change. @janstenpickle don't you mind?

Reply:
I don't mid really. I'm just thinking we can maybe preserve the old API with a deprecation notice?

Reply:
Sure, will rework

Reply:
@janstenpickle we can preserve this:
```scala
  @deprecated
  def url: String = uri.renderString
```

Would you also like to keep the old constructor for `case class Observer(url: String)`?